webalyzer/collector/static/collect.js

```diff
@@ -42,44 +53,78 @@
     };
     req.send();
   }
+
   function postCSS(url, callback) {
     _getCSS(url, function(status, response) {
       if (status === 200) {
-        var req = request();
-        var data = new FormData();
-        data.append('url', url);
-        data.append('domain', domain);
-        data.append('css', response);
-        _post(data, function(status) {
-          if (status === 200) {
-            console.log("We have already collected this CSS");
-            if (callback) callback(url, false);
-          } else if (status === 201) {
-            console.log("Yum! Collected another chunk of CSS");
-            if (callback) callback(url, true);
-          } else {
-            console.warn("Failed to collect the CSS of this page");
-          }
+        var source_hash = makeHash(response);
+
+        _postHash(source_hash, 'css', function(status) {
+            if(status === 200) {
+                console.log('We have already collected CSS matching ' + source_hash);
+            } else {
+                console.log('Sending CSS.');
+                var req = request();
+                var data = new FormData();
+                data.append('url', url);
+                data.append('domain', domain);
+                data.append('css', response);
+                data.append('source_hash', source_hash);
+                _post(data, function(status) {
+                    if (status === 200) {
+                        console.log("We have already collected this CSS");
+                    if (callback) callback(url, false);
+                    } else if (status === 201) {
+                        console.log("Yum! Collected another chunk of CSS");
+                    if (callback) callback(url, true);
```

Comment:
The indentation here is a bit off.  Same with the `if (callback) ...` line 4 lines above.